src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,112 @@
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { useEffect, useState } from 'react';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { FilterBySex } from '../types/FilterBySex';
+import { useSearchParams } from 'react-router-dom';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [isError, setIsError] = useState<boolean>(false);
+
+  const [searchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || '';
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const [activeSexFilter, setActiveSexFilter] = useState<FilterBySex>(
+    FilterBySex.ALL,
+  );
+
+  useEffect(() => {
```

Comment:
in such cases you don't need a separate state for `activeSexFilter`
It can be replaced with common variable (wrapped in useMemo):
```
const activeSexFilter = useMemo(() => {
  if (sex === 'm') {
    return FilterBySex.MALE;
  } 
  
  if (sex === 'f') {
    return FilterBySex.FEMALE;
  } 

   return FilterBySex.ALL;
}, [sex]);
```

Reply:
General rule - do not create separate state if you can calculate it using other state/props